### PR TITLE
#158457051 users should have profile page

### PR DESCRIPTION
**Pull request by Hasstrup**
#### What does this PR do?
This Pull request adds the profile page feature. Implements the api/profile endpoint and the subsequent controller and directive.
 Implements authenticated endpoint /api/profile/
  Implements directive and controller
  Subsequent tests

#### Description of Task to be completed?
'As a registered user I should have a profile page and have displayed my personal log of previously played games and number of games I have won.

*Acceptance Criteria*
GIVEN I am a registered user that has logged in
WHEN I click on the "profile" link from the main navigation
THEN I should see a profile page with my picture and details

WHEN I click on the "profile" link from the main navigation
THEN I also see a list of all games played and number of games won.

#### How should this be manually tested?
Travis build
https://travis-ci.org/andela/ET-cfh/builds/404156323?utm_source=github_status&utm_medium=notification

#### What are the relevant pivotal tracker stories?
158457051 Users should have a profile page
#### Screenshots
screen shot 2018-07-15 at 3 53 23 pm

### CODE REVIEW

File: .eslintrc 2.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+module.exports = {
```
Comment (adetola-ralph) [inline .eslintrc 2.js]: What is the use of file when we already have an .eslintrc.js file

File: app/controllers/users.js [hunk 2]
```diff
@@ -8,7 +8,10 @@ import mongoose from 'mongoose';
 import passport from 'passport';
 import avatarsList from './avatars';
 import { Tokenizer } from '../helpers/tokenizer';
-import sendInvitationEmail from '../helpers/sendInvitationEmail';
```
Comment (adetola-ralph) [inline app/controllers/users.js:11]: Why the change of name here, is it necessary?

File: backend-test/integration/auth.test.js [hunk 3]
```diff
@@ -1,66 +0,0 @@
-/* eslint no-undef: 0 */
```
Comment (adetola-ralph) [inline backend-test/integration/auth.test.js:1]: Why was this file deleted?

File: backend-test/integration/user.test.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+describe('User endpoints', () => {
+  describe('Authentication', () => {
+    before(() => Promise.resolve(User.create(mock).then((user) => {
+      token = Tokenizer(user);
+    })
+      .catch((err) => {
+        throw err;
+      })));
+
+    it('POST /api/auth/login should return the user token along with the', (done) => {
```
Comment (adetola-ralph) [inline backend-test/integration/user.test.js]: `Along with the` what?

File: backend-test/integration/user.test.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      try {
```
Comment (adetola-ralph) [inline backend-test/integration/user.test.js]: Once again try catch is redundant, please remove it. I have mentioned this a lot before

File: public/js/controllers/invitePlayers.js [hunk 6]
```diff
@@ -27,6 +27,11 @@ angular.module('mean.system')
       });
     };
 
+    $scope.closeReusedModal = function () {
+      $('#reuse-modal').modal('close');
+      game.isFilledUp = null;
```
Comment (adetola-ralph) [inline public/js/controllers/invitePlayers.js]: Why null and not false

**Review by adetola-ralph — CHANGES_REQUESTED** [review]

File: backend-test/integration/auth.test.js (hunk 3 above)
Comment (Hasstrup) [reply backend-test/integration/auth.test.js:1]: I moved all the auth tests there to user.test.js, since I also wrote tests for the user profile there

File: app/controllers/users.js (hunk 2 above)
Comment (Hasstrup) [reply app/controllers/users.js:11]: Yes, I was getting an eslint error, that was the name of the method in the module he created

File: .eslintrc 2.js (hunk 1 above)
Comment (Hasstrup) [reply .eslintrc 2.js]: will delete right away

File: backend-test/integration/user.test.js (hunk 5 above)
Comment (Hasstrup) [reply backend-test/integration/user.test.js]: Will do. Thank you

File: public/js/directives/profile.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      .then(function (response) {
+        $scope.currentUser = response.data.data;
+        $scope.currentUser.avatar = $scope.currentUser.avatar || placeholder;
+        $scope.games = response.data.data.games;
+        $scope.gamesWon = response.data.data.gamesWon;
+      }, function () {});
+  };
+
+  $scope.generateDate = function (item) {
+    return moment(new Date(item.createdAt)).fromNow();
```
[the remark this answers is not in the document]
Comment (Hasstrup) [reply public/js/directives/profile.js]: This is happening because moment is defined in the html file, in a script file in foot.jade

**Review by adetola-ralph — APPROVED** [review]
